python/sglang/srt/managers/scheduler_snapshot_handlers.py

```
@@ -0,0 +1,1123 @@
+            # new_pool_idx is a (1,)-tensor; get a 0-dim tensor and a scalar
+            new_pool_idx_0d = new_pool_idx[0]  # 0-dim tensor — assign to req
+            new_pool_idx_scalar = new_pool_idx_0d.item()  # Python int — for inject
+            pool_freed = False  # guard for outer except cleanup
+
+            # Inject state into the new pool slot
+            scheduler.snapshot_manager.inject_state_to_pool(
+                conv_states, temporal_states, mamba_pool, new_pool_idx_scalar
+            )
+
+            # Create new request ID
+            new_rid = f"restored-{uuid.uuid4().hex[:8]}"
```

Ruff (0.15.13)

[error] Undefined name `uuid`

(F821)